### PR TITLE
Superbuild cmake

### PR DESCRIPTION
**Pull request by cryos**

### CODE REVIEW

**Comment by cryos** [comment]
How does this look @utkarshayachit, @mathturtle - it does what I wanted, i.e. I can enable tomviz by default (I also pulled the latest version of the scripts from ParaView's superbuilds before applying the patch, and it seemed to work OK when configuring).

File: cmake/ParaViewModules.cmake [hunk 1]
```diff
@@ -310,12 +383,6 @@ function(add_external_project_internal name)
       LD_LIBRARY_PATH "${ld_library_path}")
   endif ()
 
-  set(ldflags_argument)
-  list(APPEND ldflags_argument LDFLAGS "${ldflags}")
-  if (SKIP_LDFLAGS_FOR_BUILD)
-    set(ldflags_argument)
-    set(SKIP_LDFLAGS_FOR_BUILD PARENT_SCOPE)
-  endif()
```
Comment (mathturtle) [inline cmake/ParaViewModules.cmake]: This block was needed to make scipy build correctly on Linux.  It hadn't migrated back to PV superbuild yet.

File: cmake/ParaViewModules.cmake [hunk 2]
```diff
@@ -329,7 +396,7 @@ function(add_external_project_internal name)
     ${${name}_revision}
 
     PROCESS_ENVIRONMENT
-      ${ldflags_argument}
```
Comment (mathturtle) [inline cmake/ParaViewModules.cmake]: See above.

File: cmake/ParaViewModules.cmake [hunk 3]
```diff
@@ -342,8 +409,10 @@ function(add_external_project_internal name)
       -DCMAKE_PREFIX_PATH:PATH=${prefix_path}
       -DCMAKE_C_FLAGS:STRING=${project_c_flags}
       -DCMAKE_CXX_FLAGS:STRING=${project_cxx_flags}
-      -DCMAKE_SHARED_LINKER_FLAGS:STRING=${ldflags}
+      -DCMAKE_SHARED_LINKER_FLAGS:STRING=${project_ld_flags}
```
Comment (mathturtle) [inline cmake/ParaViewModules.cmake]: See above

**Comment by mathturtle** [comment]
@mathstuf Did you have any changes that will be overridden by this?

**Comment by cryos** [comment]
Sounds good, I would like to minimize our divergence from ParaView, but could remove the update for now, and then work to merge our changes the other way if that is easier. I would like to avoid forking the PV scripts, and this seemed like a good opportunity to examine what was still needed in tomviz.

**Comment by mathstuf** [comment]
The diff looks good to me.

**Pull request merged**